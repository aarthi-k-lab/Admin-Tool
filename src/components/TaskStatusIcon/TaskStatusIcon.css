.smallIcon {
  max-width: 1.5rem;
  margin: 0 1rem 1rem 1rem;
  display: flex;
}

.largeIcon {
  max-width: 2rem;
  margin: 0 1rem 1rem 1rem;
  display: flex;
}
