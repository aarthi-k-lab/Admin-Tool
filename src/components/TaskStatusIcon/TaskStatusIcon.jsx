import React from 'react';
import PropTypes from 'prop-types';
import './TaskStatusIcon.css';

function getIcon(task, isSubTask, checked = false) {
  let iconName = '';
  switch (task.state) {
    case 'completed':
      iconName = isSubTask ? 'default_selected_blue_small' : 'default_selected_blue';
      break;
    case 'failed':
      iconName = isSubTask ? 'default_error_small' : 'default_error';
      break;
    default:
      iconName = task.checked || checked ? 'default_selected' : 'default_unselected';
  }
  return <img alt="task_pane_icon" src={`/static/img/${iconName}.svg`} styleName={isSubTask ? 'smallIcon' : 'largeIcon'} />;
}

function TaskStatusIcon({
  isSelected,
  task,
  isSubTask,
}) {
  return (
    getIcon(task, isSubTask, isSelected)
  );
}

TaskStatusIcon.defaultProps = {
  className: '',
  isSubTask: false,
  isSelected: false,
};

TaskStatusIcon.propTypes = {
  className: PropTypes.string,
  isSelected: PropTypes.bool,
  isSubTask: PropTypes.bool,
  task: PropTypes.shape({
    failure: PropTypes.bool,
    taskName: PropTypes.string,
  }).isRequired,
};

export { TaskStatusIcon };
export default TaskStatusIcon;
