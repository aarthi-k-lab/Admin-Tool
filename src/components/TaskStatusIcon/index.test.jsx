import React from 'react';
import renderer from 'react-test-renderer';
import TaskStatusIcon from './TaskStatusIcon';

describe('<TaskStatusIcon />', () => {
  const task = {
    taskName: 'Document and Income Type Verification',
    subTasks: [
      {
        taskName: 'Document Review',
      },
      {
        taskName: 'Choose income type',
        completed: true,
      },
      {
        taskName: 'Suspicious Activity Review',
      },
    ],
  };

  it('TaskStatusIcon renders correctly', () => {
    const tree = renderer
      .create(<TaskStatusIcon
        isSubTask={false}
        task={task}
      />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
