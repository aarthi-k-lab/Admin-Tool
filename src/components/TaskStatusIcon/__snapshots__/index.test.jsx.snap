// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TaskStatusIcon /> TaskStatusIcon renders correctly 1`] = `
<img
  alt="task_pane_icon"
  className="components-TaskStatusIcon-TaskStatusIcon__largeIcon"
  src="/static/img/default_unselected.svg"
/>
`;
